### DISCUSSION TITLE
Am I doing something wrong? Zig-zag temperature

### DISCUSSION THREAD

**Discussion by nalpo** [jmcollin78/versatile_thermostat#753] [Q&A]
Hi,

I bought a couple of Sonoff TRVZBs and have a bit of trouble controlling the temperature with them. For "self-regulation" I use "Direct control of valve" with a threshold of 0.5 and a regulation period of 5 minutes. As a sensor I use a Sonoff SNZB-02D with a resolution of 0.2° C. AFAIK I put in the correct entities for opening degree, offset and closing degree, the chosen algorithm is tpi.

I thought the algorithm would calculate and adjust the opening/closing degrees based on the temperature and try to prevent overshooting of temperature. Still, when I look at the valve open percent over time, it looks like this:

![image](https://github.com/user-attachments/assets/58ccb79f-d841-4264-8dc1-7706b5815363)

The thermostat graph tehn looks like this, with frequent overshooting:

![image](https://github.com/user-attachments/assets/f4c50131-8634-42ec-a9e9-dc0be2a045f4)

What could be the problem here, is there something wrong with my configuration? I would be gla if cou could help me with that.

**Comment by jmcollin78** [comment]
Hello @nalpo 

Look at this to tune your coefs: https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/algorithms.md#principle

The oscillation is likely due to excessive coef_int. You should decrease its value.

**Reply by nalpo** [reply]
Thanks, I'll try it with a coef_int of 0.2 and check if anything changes. I also reduced the regulation interval to 1 minute since I had the feeling that it could be a bit quicker in adjusting.

**Reply by jmcollin78** [reply]
0.2 is very low but give it a try. I should try 0.4 before.

**Reply by nalpo** [reply]
So I played around with coef_int for a while and the problem seems to be worse isntead of better:
![image](https://github.com/user-attachments/assets/3fa6ef7c-349b-4bb3-879c-c4b96e4c4d84)

For some reason, now heating starts even before reaching the set temperature (for the last heating period, coef_int was set to 0,4). Is there anything else I could try?

**Reply by nalpo** [reply]
So I wanted to give an update. I am not so sure what lead to success in the end, but here is what I changed:
- Set the regulation period to 1 minute
- Set coef_int to 0.2
- Moved my thermometer
This is how it looks now:

![image](https://github.com/user-attachments/assets/0e1f294a-9fe6-421e-af33-a41911bfe545)

So there is still an overshoot, but the heating is applied more continously now and the amount of overshooting is a lot less, 0.4-0.6 degrees instead of 1-2 before.

I think moving the thermometer had a big impact. In the room, there is a shelf and a couch in front of the radiator, the thermometer was on top of the window board right above the radiator. My windows are not insulated well and I think that played a role in not correctly recognizing the change in temperature. After moving it more towards the middle of the room, it is a lot better.

**Reply by jmcollin78** [reply]
Hello, nice changes. You can now lower the coef_ext, you are always over heating.

**Reply by nalpo** [reply]
I trietd that, but it is already at 0.01 and it seems that the interface does not allow for smaller values. Ist there a way to finetune this somehwere else?

**Reply by jmcollin78** [reply]
You can set it to 0 to try.

**Reply by nalpo** [reply]
I'll try and see how well it works. But doesn't that render the function to include external temperatures useless? Thank you though, I really appreciate it :)

**Reply by jmcollin78** [reply]
yes but you seems to not need the compensation of the outdoor temperature
